src/orders/components/OrderMetadataDialog/OrderMetadataDialog.test.tsx

```diff
@@ -0,0 +1,171 @@
+import { MockedProvider } from "@apollo/client/testing";
+import { useHasManageProductsPermission } from "@dashboard/orders/hooks/useHasManageProductsPermission";
+import { fireEvent, render, screen, within } from "@testing-library/react";
+import React from "react";
+import { IntlShape, MessageDescriptor } from "react-intl";
+
+import { OrderMetadataDialog, OrderMetadataDialogData } from "./OrderMetadataDialog";
+import { TEST_ID_ORDER_LINE_METADATA, TEST_ID_PRODUCT_VARIANT_METADATA } from "./test-ids";
+
+jest.mock("react-intl", () => ({
+  FormattedMessage: (props: MessageDescriptor) => <>{props.defaultMessage || props.id || ""}</>,
+  defineMessages: (messages: Record<string, MessageDescriptor>) => messages,
+  useIntl: (): Pick<IntlShape, "formatMessage"> => ({
+    formatMessage: ({ defaultMessage }: MessageDescriptor) => defaultMessage || "",
+  }),
+}));
+
+jest.mock("@dashboard/orders/hooks/useHasManageProductsPermission", () => ({
+  useHasManageProductsPermission: jest.fn(() => false),
+}));
+
+const mockData: OrderMetadataDialogData = {
+  id: "order-line-id",
+  productName: "Product Name",
+  metadata: [{ key: "order-line-key", value: "order-line-value", __typename: "MetadataItem" }],
+  privateMetadata: [
+    {
+      key: "order-line-private-key",
+      value: "order-line-private-value",
+      __typename: "MetadataItem",
+    },
+  ],
+  variant: {
+    id: "variant-id",
+    metadata: [{ key: "variant-key", value: "variant-value", __typename: "MetadataItem" }],
+    privateMetadata: [
+      { key: "variant-private-key", value: "variant-private-value", __typename: "MetadataItem" },
+    ],
+  },
+};
+
+describe("OrderMetadataDialog", () => {
+  const onCloseMock = jest.fn();
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it("closes when user hits close icon button", () => {
+    render(
+      <MockedProvider>
```

Comment:
suggestion: You could create wrapper component and pass it to each render

Reply:
Actually I just got rid of it altogether, I was supposed to remove it 😅